cli/templatepull_test.go

```diff
@@ -40,165 +40,218 @@ func dirSum(t *testing.T, dir string) string {
+	ctx := context.Background()
 
-		dir := t.TempDir()
+	err = extract.Tar(ctx, bytes.NewReader(expected), expectedDest, nil)
+	require.NoError(t, err)
 
-		expectedDest := filepath.Join(dir, "expected")
-		actualDest := filepath.Join(dir, "actual")
-		ctx := context.Background()
+	inv, root := clitest.New(t, "templates", "pull", template.Name, actualDest)
+	clitest.SetupConfig(t, client, root)
 
-		err = extract.Tar(ctx, bytes.NewReader(expected), expectedDest, nil)
-		require.NoError(t, err)
+	ptytest.New(t).Attach(inv)
+
+	require.NoError(t, inv.Run())
 
-		inv, root := clitest.New(t, "templates", "pull", template.Name, actualDest)
-		clitest.SetupConfig(t, client, root)
+	require.Equal(t,
+		dirSum(t, expectedDest),
+		dirSum(t, actualDest),
+	)
+}
 
-		ptytest.New(t).Attach(inv)
+// ToDir tests that 'templates pull' pulls down the latest template
+// and writes it to a directory with the name of the template if the path is not implicitly supplied.
+// nolint: paralleltest
+func TestTemplatePull_ToImplicit(t *testing.T) {
+	client := coderdtest.New(t, &coderdtest.Options{IncludeProvisionerDaemon: true})
+	user := coderdtest.CreateFirstUser(t, client)
 
-		require.NoError(t, inv.Run())
+	// Create an initial template bundle.
+	source1 := genTemplateVersionSource()
+	// Create an updated template bundle. This will be used to ensure
+	// that templates are correctly returned in order from latest to oldest.
+	source2 := genTemplateVersionSource()
 
-		require.Equal(t,
-			dirSum(t, expectedDest),
-			dirSum(t, actualDest),
-		)
-	})
+	expected, err := echo.Tar(source2)
+	require.NoError(t, err)
 
-	// FolderConflict tests that 'templates pull' fails when a folder with has
-	// existing
-	t.Run("FolderConflict", func(t *testing.T) {
-		t.Parallel()
+	version1 := coderdtest.CreateTemplateVersion(t, client, user.OrganizationID, source1)
+	_ = coderdtest.AwaitTemplateVersionJob(t, client, version1.ID)
 
-		client := coderdtest.New(t, &coderdtest.Options{IncludeProvisionerDaemon: true})
-		user := coderdtest.CreateFirstUser(t, client)
+	template := coderdtest.CreateTemplate(t, client, user.OrganizationID, version1.ID)
 
-		// Create an initial template bundle.
-		source1 := genTemplateVersionSource()
-		// Create an updated template bundle. This will be used to ensure
-		// that templates are correctly returned in order from latest to oldest.
-		source2 := genTemplateVersionSource()
+	// Update the template version so that we can assert that templates
+	// are being sorted correctly.
+	_ = coderdtest.UpdateTemplateVersion(t, client, user.OrganizationID, source2, template.ID)
 
-		expected, err := echo.Tar(source2)
+	// create a tempdir and change the working directory to it for the duration of the test (cannot run in parallel)
+	dir := t.TempDir()
+	wd, err := os.Getwd()
+	require.NoError(t, err)
+	err = os.Chdir(dir)
+	require.NoError(t, err)
+	defer func() {
+		err := os.Chdir(wd)
 		require.NoError(t, err)
```

Comment:
An additional explanation to the required might be good here since this may break other tests if it fails for whatever reason.